### PR TITLE
Login fix

### PR DESCRIPTION
**Pull request by Serhii-Buniak**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by hryhoriivoloshyn — APPROVED** [review]

**Review by vsynieok — APPROVED** [review]

File: EPlast/EPlast.WebApi/Controllers/LoginController.cs [hunk 1]
```diff
@@ -164,7 +164,7 @@ public async Task<IActionResult> Login([FromBody] LoginDto loginDto)
                     return BadRequest(_resources.ResourceForErrors["Login-InCorrectPassword"]);
                 }
             }
-            return Ok(_resources.ResourceForErrors["ModelIsNotValid"]);
+            return BadRequest(_resources.ResourceForErrors["ModelIsNotValid"]);
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/LoginController.cs:167]: Change it to `return BadRequest(ModelState);`

And, please, invert `if` statement. To reduce amount of curly braces.

File: EPlast/EPlast.WebApi/Extensions/HttpRequestExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EPlast.WebApi.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+        public static string GetFrontEndURL(this HttpRequest request, [NotNull] string queryEnd = "")
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Extensions/HttpRequestExtensions.cs:13]: Rename `queryEnd` to `tail` to reduce confusion.

Also make it nullable. If `tail` is null or empty, then just return `frontendUrl`

File: EPlast/EPlast.BLL/Handlers/PrecautionHandlers/GetUsersPrecautionsForTableHandler.cs [hunk 3]
```diff
@@ -46,10 +46,13 @@ public GetUsersPrecautionsForTableHandler(IRepositoryWrapper repositoryWrapper,
...
+                false => x => 
+                x.Number.ToString().Contains(searchedData) 
                 || (x.User.FirstName + " " + x.User.LastName).Contains(searchedData)
-                || x.Date.Date.ToString().Contains(getDate) || x.EndDate.Date.ToString().Contains(getDate)
-                || x.Reporter.Contains(searchedData) || x.Reason.Contains(searchedData)
+                || x.Date.Date.ToString().Contains(getDate) 
+                || x.Date.AddMonths(x.Precaution.MonthsPeriod).Date.ToString().Contains(getDate)
+                || x.Reporter.Contains(searchedData) 
+                || x.Reason.Contains(searchedData)
                 || x.Precaution.Name.Contains(searchedData)
```
Comment (tvardero) [inline EPlast/EPlast.BLL/Handlers/PrecautionHandlers/GetUsersPrecautionsForTableHandler.cs:56]: What is this and why is it here?

**Review by tvardero — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Handlers/PrecautionHandlers/GetUsersPrecautionsForTableHandler.cs (hunk 3 above)
Comment (Serhii-Buniak) [reply EPlast/EPlast.BLL/Handlers/PrecautionHandlers/GetUsersPrecautionsForTableHandler.cs:56]: It`s bug on precaution(search wasn't work correct).

File: EPlast/EPlast.WebApi/Controllers/LoginController.cs [hunk 4]
```diff
@@ -130,41 +130,37 @@ public async Task<IActionResult> GoogleLogin(string googleToken)
         [AllowAnonymous]
         public async Task<IActionResult> Login([FromBody] LoginDto loginDto)
         {
-            if (ModelState.IsValid)
+            if (ModelState.IsValid == false)
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/LoginController.cs:133]: `if (!ModelState.IsValid)`

File: EPlast/EPlast.WebApi/Controllers/LoginController.cs [hunk 5]
```diff
@@ -130,41 +130,37 @@ public async Task<IActionResult> GoogleLogin(string googleToken)
...
-                {
-                    return BadRequest(_resources.ResourceForErrors["Login-InCorrectPassword"]);
-                }
+                return BadRequest(ModelState);
+            }
+
+            var user = await _authService.FindByEmailAsync(loginDto.Email);
+            if (user == null)
+            {
+                return BadRequest(_resources.ResourceForErrors["Login-NotRegistered"]);
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/LoginController.cs:141]: `return NotFound();` is better suited for these situations

**Review by tvardero — APPROVED** [review]

**Review by tvardero — APPROVED** [review]

**Pull request merged**